demisto_sdk/commands/update_release_notes/update_rn_manager.py

```diff
@@ -0,0 +1,199 @@
+        filtered_added_files = filter_files_by_type(added_files, skip_file_types=SKIP_RELEASE_NOTES_FOR_TYPES)
+        if self.given_pack:  # A specific pack was chosen to update
+            self.create_pack_release_notes(self.given_pack, filtered_modified_files, filtered_added_files,
+                                           old_format_files)
+
+        elif self.changed_packs_from_git:  # update all changed packs
+            for pack in self.changed_packs_from_git:
+                if 'APIModules' in pack:  # We already handled Api Modules so we can skip it.
+                    continue
+                self.create_pack_release_notes(pack, filtered_modified_files, filtered_added_files, old_format_files)
+        else:
+            print_warning('No changes that require release notes were detected. If such changes were made, '
+                          'please commit the changes and rerun the command')
+
+    def create_pack_release_notes(self, pack: str, filtered_modified_files: set, filtered_added_files: set,
+                                  old_format_files: set):
+        """ Creates the release notes for a given pack if was changed.
+
+            :param
+                pack: The pack to create release notes for
+                filtered_modified_files: A set of filtered modified files
+                filtered_added_files: A set of filtered added files
+                old_format_files: A set of old formatted files
+        """
+        existing_rn_version = self.get_existing_rn(pack)
+        if existing_rn_version is None:  # New release notes file already found for the pack
+            sys.exit(0)
+        pack_modified = filter_files_on_pack(pack, filtered_modified_files)
+        pack_added = filter_files_on_pack(pack, filtered_added_files)
+        pack_old = filter_files_on_pack(pack, old_format_files)
+
+        # Checks if update is required
+        if pack_modified or pack_added or pack_old or self.is_force:
+            update_pack_rn = UpdateRN(pack_path=f'Packs/{pack}', update_type=self.update_type,
+                                      modified_files_in_pack=pack_modified.union(pack_old),
+                                      pre_release=self.pre_release,
+                                      added_files=pack_added, specific_version=self.specific_version,
+                                      text=self.text, is_force=self.is_force,
+                                      existing_rn_version_path=existing_rn_version)
+            updated = update_pack_rn.execute_update()
+            # If new release notes were created add it to the total number of packs that were updated.
+            if updated:
+                self.total_updated_packs.add(pack)
+                # If previous release notes existed, remove previous
+                if update_pack_rn.should_delete_existing_rn:
+                    os.unlink(self.packs_existing_rn[pack])
```

Comment:
better explain this part.
when will this happen?

Reply:
Done